cts/cts.in

```diff
@@ -156,15 +156,21 @@ if [ x$cmd != x ]; then
 	    ;;
 	trace-add|tadd)
 	    echo "Adding $trace to PCMK_trace_functions"
-	    cluster-helper -g $cluster_name -- sed -i "s/.*PCMK_trace_functions=/PCMK_trace_functions=$trace,/" $config
+	    cp "$config" "$config.$$"
+	    cluster-helper -g $cluster_name -- sed -e "s/.*PCMK_trace_functions=/PCMK_trace_functions=$trace,/" "$config" > "$config.$$"
+	    mv -- "$config.$$" "$config"
 	    ;;
 	trace-rm|trm)
 	    echo "Removing $trace from PCMK_trace_functions"
-	    cluster-helper -g $cluster_name -- sed -i "s/.*PCMK_trace_functions=\\\\\\(.*\\\\\\)$trace,\\\\\\(.*\\\\\\)/PCMK_trace_functions=\\\\\\1\\\\\\2/" $config
+	    cp "$config" "$config.$$"
+	    cluster-helper -g $cluster_name -- sed -e "s/.*PCMK_trace_functions=\\\\\\(.*\\\\\\)$trace,\\\\\\(.*\\\\\\)/PCMK_trace_functions=\\\\\\1\\\\\\2/" "$config" > "$config.$$"
```

Comment:
what the heck is this pattern doing ...

Reply:
If I were to guess, I would say it's trying to remove whatever functions are in $trace from the PCMK_trace_functions= list.  But because it's in one shell script, that gets passed as an argument to another shell script, that gets passed through ssh to a shell on a remote host, to sed, everything has to be escaped a thousand times.

I guess my first question would be:  is anyone actually using this?  And then if so, maybe we should break this out into a tiny script that gets installed on the remote system, and that gets called and we can skip all this ridiculous escaping.

Reply:
I vaguely recall seeing some use of the cluster-* scripts reported upstream. Probably worth keeping around for now. If the pattern works, I'd say just leave it alone